### PR TITLE
Remove Theme & Modules Catalog and PrestaTrust calls

### LINKED ISSUE #24639: Remove marketplace from core
#### Is your feature request related to a problem?

This Feature is a part of split between corporate prestashop/Open-sour prestashop.
Goal is to provide a full "non corporate" project.

Remove the module & theme catalog
Remove update API

#### Describe the solution you'd like

A clear and concise description of what you'd want to happen.
Remove all the call and implement an empty skeleton for the new API 

#### Tasks

- [x] #24739
- [x] #24740
- [x] #24741
- [x] #24742
- [x] #24743
- [x] #24744
- [x] #24746
- [x] #24747
- [x] #24748
- [x] #24749
- [x] #24750
- [x] #24751
- [x] #24752
- [x] #25028
- [x] #25030
- [x] #25033
- [x] #25008
- [x] #25457
- [x] Docs
  - PrestaShop/docs#1108
  - PrestaShop/docs#1088

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove occurencies from addons and prestatrust
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Related to #24639
| How to test?      | Theme and Module catalog are no longer available. You're now able to install modules from the Module Manager section. PrestaTrust has been deleted, and everything around it. Have a look at the dashboard.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

## BC breaks

 - `PrestaShop\PrestaShop\Adapter\Module\PrestaTrust` namespace has been deleted
 - `PrestaShop\PrestaShop\Core\Addon\Module::setPrestaTrustChecker` has been deleted
 - `PrestaShopBundle\Controller\Admin\Improve\Design\ThemeCatalogController` has been deleted
- `PrestaShopBundle\Controller\Admin\Improve\ModuleController::catalogAction` has been deleted
- `PrestaShopBundle\Controller\Admin\Improve\ModuleController::getPreferredModulesAction` has been deleted
- `PrestaShopBundle\Controller\Admin\Improve\ModuleController::refreshCatalogAction` has been deleted
- `PrestaShopBundle\Controller\Admin\Improve\Modules\AddonsStoreController` has been deleted
- `PrestaShopBundle\Form\Admin\Type\AddonsConnectType` has been deleted
- `admin_module_catalog`, `admin_module_catalog_refresh`, `admin_module_catalog_post` and `admin_module_addons_store` routes have been deleted

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25245)

### CODE REVIEW

**Review by matks — DISMISSED** [review]
I suggest we run e2e tests on this PR as part of the How to test

File: admin-dev/themes/default/js/bundle/module/module_card.js [hunk 1]
```diff
@@ -57,62 +57,6 @@ const AdminModuleCard = function () {
...
-    modal.find('#pstrust-img').attr({src: module.img, alt: module.name});
-    modal.find('#pstrust-name').text(module.displayName);
-    modal.find('#pstrust-author').text(module.author);
-    modal.find('#pstrust-label').attr('class', `text-${alertClass}`).text(module.prestatrust.status ? 'OK' : 'KO');
-    modal.find('#pstrust-message').attr('class', `alert alert-${alertClass}`);
-    modal.find('#pstrust-message > p').text(module.prestatrust.message);
-
-    // eslint-disable-next-line
-    return modal;
-  };
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/module/module_card.js]:
```suggestion
```

File: tests/Integration/PrestaShopBundle/Controller/Admin/Improve/ModuleControllerTest.php [hunk 2]
```diff
@@ -125,14 +125,15 @@ public function testImportModuleAction(): void
         $this->assertEquals('This functionality has been disabled.', $decodedContent['msg']);
     }
 
-    public function testRecommendedModules(): void
+    /**
+     * @return string
+     */
+    protected function getExpectedErrorMessage()
```
Comment (Progi1984) [inline tests/Integration/PrestaShopBundle/Controller/Admin/Improve/ModuleControllerTest.php]: Why this code has been added ?

**Review by Progi1984 — DISMISSED** [review]
@PierreRambaud Thank you for your contributions. As Addons has been removed, in `src/Adapter/Employee/EmployeeFormAccessChecker.php`, you can remove `$this->tabRepository` and the parameter in constructror like I do in #25559

File: tests/Integration/PrestaShopBundle/Controller/Admin/Improve/ModuleControllerTest.php (hunk 2 above)
Comment (PierreRambaud) [reply tests/Integration/PrestaShopBundle/Controller/Admin/Improve/ModuleControllerTest.php]: It was a test, forget to remove it

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PierreRambaud,

Could you please check the Conflicting files?

Thanks!

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by marionf** [comment]
I put the PR on waiting for PM, we will discuss it with @MatShir
If the Module catalog page is deleted there could be a problem for those who will use MBO: they will see the modules present on the disk and not installed both in Module Manager and in Module Catalog

**Comment by PierreRambaud** [comment]
> I put the PR on waiting for PM, we will discuss it with @MatShir If the Module catalog page is deleted there could be a problem for those who will use MBO: they will see the modules present on the disk and not installed both in Module Manager and in Module Catalog

I don't get it, what MBO is doing here :sweat_smile: 
Mbo shouldn't display modules present on disk. It's a catalog that represents what offer the addons marketplace, not the disk.

**Comment by marionf** [comment]
If you use MBO you will see some uninstalled module in Module Catalog page and some in Module Manager, it could be confusing for users

**Comment by PierreRambaud** [comment]
> If you use MBO you will see some uninstalled module in Module Catalog page and some in Module Manager, it could be confusing for users

As MBO have to completely refactor the Module Catalog behavior and Module themes, this is something we can remove from MBO.
It's a MBO problem, not a Core problem.
Also, adding modules present on disk in the addons catalog has always been a mess for the community and developpers :(

**Review by jolelievre — DISMISSED** [review]
Thanks @PierreRambaud

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 3]
```diff
@@ -477,7 +463,16 @@ class AdminModuleController {
 
         // Check for same status
         if (self.currentRefStatus !== null) {
-          isVisible &= currentModule.active === self.currentRefStatus;
+          isVisible &= (
+            (
+              currentModule.active === self.currentRefStatus
+                && currentModule.installed === true
```
Comment (sowbiba) [inline admin-dev/themes/new-theme/js/pages/module/controller.js:469]: 1 or true ?

**Review by kpodemski — DISMISSED** [review]

**Review by sowbiba — DISMISSED** [review]
LGTM. Just one comparison value to check

File: admin-dev/themes/new-theme/js/pages/module/controller.js (hunk 3 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js:469]: true, it's a boolean as it's defined on line 327 :thinking:

Comment (sowbiba) [reply admin-dev/themes/new-theme/js/pages/module/controller.js:469]: sorry I read installed = 1

**Comment by Robin-Fischer-PS** [comment]
Hi @PierreRambaud !

As seen together, I have only one error, not related to this PR : 

The "Theme Catalog" button in "Theme & Logo" page is still there, it will be removed in another PR.

pr25245 theme catalog button

Otherwise, it's OK ! QA :heavy_check_mark:

**Comment by Progi1984** [comment]
@Robin-Fischer-PS I remove the "QA :heavy_check_mark:" as the PR is in Waiting for PM since 2 days. (Ping @marionf)

**Comment by marionf** [comment]
Good, for me I am going to do an issue with the things to correct ;)

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by marionf** [comment]
Here is the issue: https://github.com/PrestaShop/PrestaShop/issues/26144

**Pull request merged**